File: personal/FormPersonal.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

"""
    Nombre: FormPersonal.py
    Autor: Lic. Claudio Invernizzi
    Fecha: 15/10/25
    E-Mail: cinvernizzi@dsgestion.site
    Proyecto: Presupuesto
    Comentarios: Clase que define el formulario de datos 
                 personales

"""

# importamos las librerías 
from clases.fuentes import Fuentes
from personal.EventosPersonal import EventosPersonal
from PySide6.QtWidgets import QDialog, QHBoxLayout, QVBoxLayout, QLabel, QLineEdit, QPushButton
from PySide6 import QtCore
from PySide6 import QtGui
from pathlib import Path


class FormPersonal(QDialog):
    """
    
        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        Definición de la clase

    """

    def __init__(self, parent):
        """
        
        :author: Claudio Invernizzi <cinvernizzi@dsgestion.site>

        :param: parent el formulario padre

        Constructor de la clase, recibe como parámetro el formulario 
        padre


        """

        # instanciamos la fuente
        fuente = Fuentes()

        # instanciamos los eventos 
        self.Eventos = EventosPersonal(self)

        # instanciamos el formulario y lo fijamos como modal
        super(FormPersonal, self).__init__(parent)

        # fijamos el tamaño y el ícono 
        self.setGeometry(200, 200, 700, 250)
        self.setWindowIcon(QtGui.QIcon('recursos/logo.png'))

        # lo establecemos como modal
        self.setModal(True)

        # definimos el layout
        formulario = QVBoxLayout()

        # agregamos el título
        lTitulo = QLabel("Datos Personales")
        lTitulo.setMaximumHeight(30)
        lTitulo.setFont(fuente.negrita)
        lTitulo.setAlignment(QtCore.Qt.AlignCenter)
        formulario.addWidget(lTitulo)

        # abrimos la primer fila 
        fila1 = QHBoxLayout()

        # presenta la clave 
        lId = QLabel("Id:")
        lId.setFont(fuente.normal)
        fila1.addWidget(lId)
        self.tId = QLineEdit()
        self.tId.setFont(fuente.normal)
        self.tId.setToolTip("Clave del registro")
        self.tId.setMaximumWidth(40)
        self.tId.setMinimumHeight(30)
        self.tId.setReadOnly(True)
        fila1.addWidget(self.tId)

        # presenta el nombre 
        lNombre = QLabel("Nombre:")
        lNombre.setFont(fuente.normal)
        fila1.addWidget(lNombre)
        self.tNombre = QLineEdit()
        self.tNombre.setFont(fuente.normal)
        self.tNombre.setToolTip("Su nombre como aparecerá en el presupuesto")
        self.tNombre.setMinimumWidth(200)
        self.tNombre.setMinimumHeight(30)
        fila1.addWidget(self.tNombre)

        # presenta la empresa
        lEmpresa = QLabel("Empresa:")
        lEmpresa.setFont(fuente.normal)
        fila1.addWidget(lEmpresa)
        self.tEmpresa = QLineEdit()
        self.tEmpresa.setFont(fuente.normal)
        self.tEmpresa.setToolTip("La empresa como aparecerá en el presupuesto")
        self.tEmpresa.setMinimumWidth(200)
        self.tEmpresa.setMinimumHeight(30)
        fila1.addWidget(self.tEmpresa)

        # agregamos la fila 
        formulario.addLayout(fila1)

        # abrimos la segunda fila 
        fila2 = QHBoxLayout()

        # presenta la dirección 
        lDireccion = QLabel("Direccion:")
        lDireccion.setFont(fuente.normal)
        fila2.addWidget(lDireccion)
        self.tDireccion = QLineEdit()
        self.tDireccion.setFont(fuente.normal)
        self.tDireccion.setToolTip("El domicilio como aparecerá en el presupuesto")
        self.tDireccion.setMinimumWidth(200)
        self.tDireccion.setMinimumHeight(30)
        fila2.addWidget(self.tDireccion)

        # presenta el cuil 
        lCuil = QLabel("Id Tributaria:")
        lCuil.setFont(fuente.normal)
        fila2.addWidget(lCuil)
        self.tCuil = QLineEdit()
        self.tCuil.setFont(fuente.normal)
        self.tCuil.setToolTip("Su identificación tributaria")
        self.tCuil.setMinimumWidth(120)
        self.tCuil.setMinimumHeight(30)
        fila2.addWidget(self.tCuil)

        # agregamos la segunda fila 
        formulario.addLayout(fila2)

        # define la tercer fila 
        fila3 = QHBoxLayout()

        # presenta el teléfono 
        lTelefono = QLabel("Teléfono:")
        lTelefono.setFont(fuente.normal)
        fila3.addWidget(lTelefono)
        self.tTelefono = QLineEdit()
        self.tTelefono.setFont(fuente.normal)
        self.tTelefono.setToolTip("Teléfono del cliente")
        self.tTelefono.setMaximumWidth(150)
        self.tTelefono.setMinimumHeight(30)
        fila3.addWidget(self.tTelefono)

        # presenta el mail 
        lMail = QLabel("E-Mail:")
        lMail.setFont(fuente.normal)
        fila3.addWidget(lMail)
        self.tMail = QLineEdit()
        self.tMail.setFont(fuente.normal)
        self.tMail.setToolTip("Dirección de correo electrónico")
        self.tMail.setMinimumHeight(30)
        fila3.addWidget(self.tMail)

        # agrega la segunda fila 
        formulario.addLayout(fila3)

        # define la tercer fila 
        fila4 = QHBoxLayout()

        # agregamos un espaciador
        fila4.addStretch()

        # verificamos si existe el logo
        file_path = "temp/logo.png"
        if Path(file_path).is_file():
            icon3 = QtGui.QIcon()
            icon3.addPixmap(QtGui.QPixmap("temp/logo.png"))            
        else:
            icon3 = QtGui.QIcon()
            icon3.addPixmap(QtGui.QPixmap("recursos/imagen_no_disponible.gif"))

        # presenta el logo 
        self.btnLogo = QPushButton()
        self.btnLogo.setFixedHeight(60)
        self.btnLogo.setFixedWidth(60)
        self.btnLogo.setIcon(icon3)
        self.btnLogo.setIconSize(QtCore.QSize(60,60))
        self.btnLogo.setToolTip("Pulse para actualizar su logo")
        fila4.addWidget(self.btnLogo)

        # asignamos el evento
        self.btnLogo.clicked.connect(self.Eventos.seleccionarLogo)

        # presenta la fecha 
        lAlta = QLabel("Alta:")
        lAlta.setFont(fuente.normal)
        fila4.addWidget(lAlta)
        self.tAlta = QLineEdit()
        self.tAlta.setFont(fuente.normal)
        self.tAlta.setToolTip("Fecha de alta del registro")
        self.tAlta.setMaximumWidth(90)
        self.tAlta.setMinimumHeight(30)
        self.tAlta.setReadOnly(True)
        fila4.addWidget(self.tAlta)

        # presenta el botón secciones
        self.btnSecciones = QPushButton("Secciones")
        self.btnSecciones.setFixedHeight(30)
        self.btnSecciones.setFixedWidth(120)
        self.btnSecciones.setToolTip("Edita las secciones de un presupuesto")
        icon4 = QtGui.QIcon()
        icon4.addPixmap(QtGui.QPixmap("recursos/secciones.png"))
        self.btnSecciones.setIcon(icon4)
        fila4.addWidget(self.btnSecciones)

        # presenta el botón grabar
        self.btnGrabar = QPushButton("Grabar")
        self.btnGrabar.setFixedHeight(30)
        self.btnGrabar.setFixedWidth(120)
        icon1 = QtGui.QIcon()
        icon1.addPixmap(QtGui.QPixmap("recursos/grabar.png"))
        self.btnGrabar.setIcon(icon1)
        self.btnGrabar.setToolTip("Graba el registro en la base")
        fila4.addWidget(self.btnGrabar)

        # asignamos el evento
        self.btnGrabar.clicked.connect(self.Eventos.verificaPersonal)

        # presenta el botón cancelar 
        self.btnCancelar = QPushButton("Cancelar")
        self.btnCancelar.setFixedWidth(120)
        self.btnCancelar.setFixedHeight(30)
        icon2 = QtGui.QIcon()
        icon2.addPixmap(QtGui.QPixmap("recursos/cancelar.png"))
        self.btnCancelar.setIcon(icon2)
        self.btnCancelar.setToolTip("Cierra el formulario")
        fila4.addWidget(self.btnCancelar)

        # asignamos el evento
        self.btnCancelar.clicked.connect(self.Eventos.cancelaPersonal)

        # agregamos la tercer fila 
        formulario.addLayout(fila4)

        # fijamos el layout 
        self.setLayout(formulario)

        # mostramos el formulario 
        self.show()

        # cargamos el registro 
        self.Eventos.cargaPersonal()
        

File: src/main/java/site/dsgestion/personal/FormPersonal.java
/**
 * 
 * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
 * @date 24/01/2026
 * @name EventosGrillaClientes
 * @Projecto: Presupuesto
 * @Copyright DsGestion <dsgestion.site>
 * @Licence: GPL
 * Clase define el formulario de los datos personales
 * 
 */

// definición del paquete
package site.dsgestion.personal;

// importamos las librerías
import java.awt.Dimension;
import java.awt.FlowLayout;
import java.awt.Image;
import java.awt.Toolkit;
import javax.swing.BoxLayout;
import javax.swing.Icon;
import javax.swing.ImageIcon;
import javax.swing.JButton;
import javax.swing.JDialog;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.JTextField;
import site.dsgestion.dbApi.Fuentes;

/**
 *
 * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
 * 
 * Definición de la clase
 * 
 */
public final class FormPersonal extends JDialog {
    
    // declaración de variables
    public JTextField tId;               // clave del registro
    public JTextField tNombre;           // nombre del usuario
    public JTextField tEmpresa;          // nombre fantasía de la empresa
    public JTextField tDireccion;        // dirección postal
    public JTextField tCuil;             // clave tributaria
    public JTextField tTelefono;         // número de teléfono
    public JTextField tMail;             // correo electrónico
    public JTextField tFecha;            // fecha de alta del registro
    public JButton btnLogo;              // botón con el logo
    protected EventosPersonal Eventos;   // clase de eventos
    
    /**
     * 
     * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
     * 
     * Constructor de la clase
     * 
     */
    public FormPersonal(){

        // instanciamos la clase de eventos
        this.Eventos = new EventosPersonal(this);

        // fijamos el tamaño
        this.setSize(850, 250);

        // podemos redimensionar
        this.setResizable(true);

        // fijamos el título
        this.setTitle("Datos Personales");
        
        // centramos la ventana
        Toolkit miPantalla = Toolkit.getDefaultToolkit();
        Dimension tamanioPantalla = miPantalla.getScreenSize();

        // ahora lo convertimos a alto y ancho
        int ancho = tamanioPantalla.width;
        int alto = tamanioPantalla.height;

        // y ahora lo movemos para centrarlo
        this.setLocation((ancho - 800)/2, (alto - 350)/2);

        // fijamos el ícono
        Image icono = new ImageIcon(getClass().getResource("/imagenes/logo.png")).getImage();
        this.setIconImage(icono);

        // configuramos la interfaz
        this.setupUi();
        
    }
    
    /**
     * 
     * @author Claudio Invernizzi <cinvernizzi@dsgestion.site>
     * 
     * Método llamado desde el constructor que configura la 
     * interfaz
     * 
     */
    protected void setupUi(){
    
        // definimos la fuente
        Fuentes Fuente = new Fuentes();
        
        // agregamos un panel contenedor
        JPanel contenedor = new JPanel();
        contenedor.setLayout(new BoxLayout(contenedor, BoxLayout.Y_AXIS));
        
        // presenta el título 
        JLabel lTitulo = new JLabel("Datos Personales");
        lTitulo.setFont(Fuente.Negrita);
        contenedor.add(lTitulo);
        
        // define la primer fila 
        JPanel fila1 = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 5));
        
        // presenta la clave del registro
        JLabel lId = new JLabel("Id:");
        lId.setFont(Fuente.Normal);
        fila1.add(lId);
        this.tId = new JTextField();
        this.tId.setFont(Fuente.Normal);
        this.tId.setToolTipText("Clave única del registro");
        this.tId.setPreferredSize(new Dimension(40,30));
        this.tId.setEditable(false);
        fila1.add(this.tId);
        
        // pide el nombre
        JLabel lNombre = new JLabel("Nombre:");
        lNombre.setFont(Fuente.Normal);
        fila1.add(lNombre);
        this.tNombre = new JTextField();
        this.tNombre.setFont(Fuente.Normal);
        this.tNombre.setToolTipText("Nombre como aparecerá en el presupuesto");
        this.tNombre.setPreferredSize(new Dimension(200,30));
        fila1.add(this.tNombre);
        
        // pide el nombre de la empresa
        JLabel lEmpresa = new JLabel("Empresa:");
        lEmpresa.setFont(Fuente.Normal);
        fila1.add(lEmpresa);
        this.tEmpresa = new JTextField();
        this.tEmpresa.setFont(Fuente.Normal);
        this.tEmpresa.setToolTipText("Nombre como aparecerá en el presupuesto");
        this.tEmpresa.setPreferredSize(new Dimension(200,30));
        fila1.add(this.tEmpresa);
        
        // pide la clave tributaria 
        JLabel lIdTributaria = new JLabel("Id Tributaria:");
        lIdTributaria.setFont(Fuente.Normal);
        fila1.add(lIdTributaria);
        this.tCuil = new JTextField();
        this.tCuil.setFont(Fuente.Normal);
        this.tCuil.setToolTipText("Indique su clave tributaria");
        this.tCuil.setPreferredSize(new Dimension(130,30));
        fila1.add(this.tCuil);
        
        // agrega la fila al contenedor
        contenedor.add(fila1);
        
        // define la segunda fila 
        JPanel fila2 = new JPanel(new FlowLayout(FlowLayout.LEFT, 5, 5));
        
        // pide la dirección 
        JLabel lDireccion = new JLabel("Dirección:");
        lDireccion.setFont(Fuente.Normal);
        fila2.add(lDireccion);
        this.tDireccion = new JTextField();
        this.tDireccion.setFont(Fuente.Normal);
        this.tDireccion.setToolTipText("Dirección que figura en el presupuesto");
        this.tDireccion.setPreferredSize(new Dimension(200, 30));
        fila2.add(this.tDireccion);

        // pide el teléfono 
        JLabel lTelefono = new JLabel("Teléfono:");
        lTelefono.setFont(Fuente.Normal);
        fila2.add(lTelefono);
        this.tTelefono = new JTextField();
        this.tTelefono.setFont(Fuente.Normal);
        this.tTelefono.setToolTipText("Número de teléfono que figura en el presupuesto");
        this.tTelefono.setPreferredSize(new Dimension(120,30));
        fila2.add(this.tTelefono);
        
        // pide el mail 
        JLabel lMail = new JLabel("Mail:");
        lMail.setFont(Fuente.Normal);
        fila2.add(lMail);
        this.tMail = new JTextField();
        this.tMail.setFont(Fuente.Normal);
        this.tMail.setToolTipText("Su dirección de correo electrónico");
        this.tMail.setPreferredSize(new Dimension(200,30));
        fila2.add(this.tMail);
        
        // presenta la fecha de alta
        JLabel lAlta = new JLabel("Alta:");
        lAlta.setFont(Fuente.Normal);
        fila2.add(lAlta);
        this.tFecha = new JTextField();
        this.tFecha.setFont(Fuente.Normal);
        this.tFecha.setToolTipText("Fecha de alta del registro");
        this.tFecha.setPreferredSize(new Dimension(90,30));
        this.tFecha.setEditable(false);
        fila2.add(this.tFecha);
        
        // agrega la segunda fila 
        contenedor.add(fila2);
        
        // define la tercer fila 
        JPanel fila3 = new JPanel(new FlowLayout(FlowLayout.RIGHT, 5, 5));        
        
        // presenta al botón con el logo
        this.btnLogo = new JButton();
        this.btnLogo.setFont(Fuente.Normal);
        this.btnLogo.setToolTipText("Pulse para cargar su logo");
        this.btnLogo.setPreferredSize(new Dimension(60,60));
        Icon icon1 = new ImageIcon(getClass().getResource("/imagenes/imagen_no_disponible.png"));
        this.btnLogo.setIcon(icon1);
        this.btnLogo.addActionListener(e -> Eventos.cargaLogo());
        fila3.add(this.btnLogo);
        
        // verifica si existe el logo
        this.Eventos.verificaLogo();
        
        // presenta el botón de secciones
        JButton btnSecciones = new JButton("Secciones");
        btnSecciones.setFont(Fuente.Normal);
        btnSecciones.setToolTipText("Configura las secciones del presupuesto");
        btnSecciones.setPreferredSize(new Dimension(120, 30));
        Icon icon2 = new ImageIcon(getClass().getResource("/imagenes/secciones.png"));
        btnSecciones.setIcon(icon2);
        btnSecciones.addActionListener(e -> Eventos.verSecciones());        
        fila3.add(btnSecciones);
        
        // el botón grabar
        JButton btnGrabar = new JButton("Grabar");
        btnGrabar.setFont(Fuente.Normal);
        btnGrabar.setToolTipText("Graba el registro en la base");
        btnGrabar.setPreferredSize(new Dimension(120, 30));
        Icon icon3 = new ImageIcon(getClass().getResource("/imagenes/grabar.png"));
        btnGrabar.setIcon(icon3);
        btnGrabar.addActionListener(e -> Eventos.verificaPersonal());        
        fila3.add(btnGrabar);
        
        // el botón cancelar
        JButton btnCancelar = new JButton("Cancelar");
        btnCancelar.setFont(Fuente.Normal);
        btnCancelar.setToolTipText("Reinicia el formulario");
        btnCancelar.setPreferredSize(new Dimension(120, 30));
        Icon icon4 = new ImageIcon(getClass().getResource("/imagenes/cancelar.png"));
        btnCancelar.setIcon(icon4);
        btnCancelar.addActionListener(e -> Eventos.cancelaPersonal());
        fila3.add(btnCancelar);
        
        // agrega la tercer fila
        contenedor.add(fila3);

        // agregamos el panel
        this.add(contenedor);
        
        // mostramos el formulario
        this.setVisible(true);
        
    }
    
}
